### PR TITLE
Fix map ui

### PR DESCRIPTION
**Pull request by talbenasuli**

### CODE REVIEW

File: Dizzy/Screens/Search Place/PlaceSearchVM.swift [hunk 1]
```diff
@@ -8,8 +8,10 @@
 
 import Foundation
 
-protocol MapSearchVMType {
-    var delegate: MapSearchVMDelegate? { get set }
+protocol PlaceSearchVMType {
+    var delegate: PlaceSearchVMDelegate? { get set }
+    var currentLocation: Observable<Location?> { get }
+
```
Comment (menesh2) [inline Dizzy/Screens/Search Place/PlaceSearchVM.swift:14]: remove one line, swift lint does not allow tow lines in space

File: Dizzy/Screens/Map/MapVM.swift [hunk 2]
```diff
@@ -73,9 +83,19 @@ final class MapVM: MapVMType {
...
+            if let url = URL(string: place.imageURLString ?? "") {
+                placeImageView.setImage(from: url)
+            }
+            
+            let mark = Mark(title: place.name, snippet: place.description, location: place.location, displayView: placeImageView)
+            return mark
+        })
     }
     
+
```
Comment (menesh2) [inline Dizzy/Screens/Map/MapVM.swift:103]: same here, remove one line

File: Dizzy/Screens/Map/MapCoordinator.swift [hunk 3]
```diff
@@ -56,24 +56,27 @@ extension MapCoordinator: MapVMDelegate {
     }
 
     func searchButtonPressed() {
-        guard var viewModel = container?.resolve(MapSearchVMType.self),
-            let searchVC = container?.resolve(MapSearchVC.self, argument: viewModel) else {
-                print("could not create MapSearchVC page")
-                return
+        guard let presentingViewController = navigationController.topViewController,
+            let searchCoordinator = container?.resolve(SearchPlaceCoordinatorType.self, argument: presentingViewController) else {
```
Comment (menesh2) [inline Dizzy/Screens/Map/MapCoordinator.swift:60]: why do you open new coordinator, the search is pard of the maps story

File: Dizzy/Screens/Map/MapVC.swift [hunk 4]
```diff
@@ -32,16 +35,34 @@ class MapVC: ViewController {
...
+        view.addSubview(currentLocationButton)
+    }
+    
+    private func setupViews() {
+        setupCurrentLocationButton()
+    }
+    
+    private func setupCurrentLocationButton() {
+        currentLocationButton.isHidden = true
+        currentLocationButton.setImage(UIImage(named: "current_location_icon"), for: .normal)
```
Comment (menesh2) [inline Dizzy/Screens/Map/MapVC.swift:49]: add `.localized` to the string

File: Dizzy/Screens/Search Place/PlaceSearchVM.swift [hunk 5]
```diff
@@ -18,18 +20,29 @@ protocol MapSearchVMType {
...
 }
 
-protocol MapSearchVMDelegate: class {
+protocol PlaceSearchVMDelegate: class {
     func didSelect(place: PlaceInfo)
     func cancelButtonPressed()
 }
 
-final class MapSearchVM: MapSearchVMType {
+final class PlaceSearchVM: PlaceSearchVMType {
```
Comment (menesh2) [inline Dizzy/Screens/Search Place/PlaceSearchVM.swift:28]: i think you should insert the search code in to `PlacesVM` and use the existing instance of it (make `PlacesVM` singleton in the assembly), this way we will have the search ability on the `discoveryVC` too

File: Dizzy/Screens/Search Place/SearchPlaceCoordinator.swift [hunk 6]
```diff
@@ -0,0 +1,59 @@
+//
```
Comment (menesh2) [inline Dizzy/Screens/Search Place/SearchPlaceCoordinator.swift:1]: as i mentioned above, i don't think we need to have new coordinator for the search

**Review by menesh2 — CHANGES_REQUESTED** [review]
please fix the crush as we talked about

File: Dizzy/Screens/Map/MapVM.swift (hunk 2 above)
Comment (talbenasuli) [reply Dizzy/Screens/Map/MapVM.swift:103]: Done

File: Dizzy/Screens/Map/MapCoordinator.swift (hunk 3 above)
Comment (talbenasuli) [reply Dizzy/Screens/Map/MapCoordinator.swift:60]: No search could be from discovery as well, so I made a coordinator for that.

File: Dizzy/Screens/Map/MapVC.swift (hunk 4 above)
Comment (talbenasuli) [reply Dizzy/Screens/Map/MapVC.swift:49]: What string?

**Pull request merged**